evap/staff/tests/test_tools.py

```diff
@@ -219,6 +222,52 @@ def test_do_nothing_if_test_run(self):
         self.assertEqual(len(messages), 4)
 
 
+class RemoveUserDueToInactivity(TestCase):
+    def test_remove_user_due_to_inactivity(self):
+        inactive_user = baker.make(UserProfile, last_name="Inactive User")
+        inactive_evaluation = baker.make(
+            Evaluation,
+            state=Evaluation.State.PUBLISHED,
+            vote_start_datetime=datetime.today()
+            - timedelta(days=settings.PARTICIPATION_DELETION_AFTER_INACTIVE_MONTHS + 1),
+            vote_end_date=datetime.today() - timedelta(days=settings.PARTICIPATION_DELETION_AFTER_INACTIVE_MONTHS + 1),
+            participants=[inactive_user],
+        )
+
+        semester = inactive_evaluation.course.semester
+        semester.archive()
+
+        messages = remove_inactive_participations(inactive_user)
+        self.assertTrue(inactive_user.can_be_marked_inactive_by_manager)
+        self.assertEqual(len(messages), 1)
+
+    def test_do_not_remove_user_due_to_inactivity(self):
+        active_user = baker.make(UserProfile, last_name="Active User")
+
+        baker.make(Evaluation, state=Evaluation.State.PUBLISHED, participants=[active_user])
+
+        messages = remove_inactive_participations(active_user)
+        self.assertFalse(active_user.can_be_marked_inactive_by_manager)
+        self.assertEqual(len(messages), 0)
+
+    def test_do_nothing_if_test_run(self):
+        inactive_user = baker.make(UserProfile, last_name="Test Run User")
+        inactive_evaluation = baker.make(
+            Evaluation,
+            state=Evaluation.State.PUBLISHED,
+            vote_start_datetime=datetime.today()
+            - timedelta(days=settings.PARTICIPATION_DELETION_AFTER_INACTIVE_MONTHS + 1),
+            vote_end_date=datetime.today() - timedelta(days=settings.PARTICIPATION_DELETION_AFTER_INACTIVE_MONTHS + 1),
+            participants=[inactive_user],
+        )
```

Comment:
Since all three tests need a user and an evaluation, it would make sense to use the [setup logic that the test framework provides](https://docs.djangoproject.com/en/5.1/topics/testing/tools/#django.test.TestCase.setUpTestData) here. If you add a method
```python3
def setUpTestData(cls):
    cls.user = baker.make(UserProfile)
    six_months_ago = datetime.today() - datetime.timedelta(days=6*30)
    cls.evaluation = baker.make(
        Evaluation,
        state=Evaluation.State.PUBLISHED,
        vote_start_datetime=six_months_ago,
        vote_end_date=six_months_ago.date() + timedelta(days=10)
        participants=[cls.user]
    )
```
you already have `self.user` and `self.evaluation` available in all test methods. You can then
* either change and save the evaluation to make it be before/after the "cutoff" date, or
* vary the settings value of `PARTICIPATION_DELETION_AFTER_INACTIVE_MONTHS ` to make that evaluation be before or after the cutoff. Django provides [`override_settings`](https://docs.djangoproject.com/en/5.1/topics/testing/tools/#django.test.override_settings) to do that. Check out other usages on the codebase (or feel free to just ask someone in person).

Reply:
Tried to implement it with override_settings, please have a look and give me some feedback if you see any improvements to be made =)